packages/jest-environment-jsdom/src/index.js

```diff
@@ -18,15 +18,20 @@ const ModuleMocker = require('jest-mock');
 
 class JSDOMEnvironment {
 
+  console: Object;
   document: ?Object;
   fakeTimers: ?FakeTimers;
   global: ?Global;
   moduleMocker: ?ModuleMocker;
 
-  constructor(config: Config): void {
+  constructor(config: Config, createJestConsole: (x: number) => Object): void {
     // lazy require
-    this.document = require('jsdom').jsdom(/* markup */undefined, {
+    const jsdom = require('jsdom');
+    const con = this.console = createJestConsole(8);
```

Comment:
You're assigning to `this.console`. Be sure to remove the references to this field in `dispose()` method, as this constructor is called for every test file (same for `jest-environment-node`) and may cause memory leak.